### ISSUE TITLE
Autocomplete does not find suggestions in polars namespaces

### ISSUE THREAD

**Issue by kjgoodrick** [marimo-team/marimo#3302]
### Describe the bug

When using polars the marimo autocomplete is unable to find any suggestions for names that are within namespaces. For example, here is what the autocomplete shows for the `dt` namespace.

![Image](https://github.com/user-attachments/assets/71169f44-6c31-4da1-8109-b2feb3c8d41a)

Here is what VS Code shows:
![Image](https://github.com/user-attachments/assets/df01004c-1bb8-495d-94a0-131240dfad10)

### Environment

{
  "marimo": "0.10.7",
  "OS": "Darwin",
  "OS Version": "24.2.0",
  "Processor": "arm",
  "Python Version": "3.13.1",
  "Binaries": {
    "Browser": "--",
    "Node": "v23.5.0"
  },
  "Dependencies": {
    "click": "8.1.3",
    "docutils": "0.21.2",
    "itsdangerous": "2.2.0",
    "jedi": "0.19.2",
    "markdown": "3.7",
    "narwhals": "1.19.1",
    "packaging": "24.2",
    "psutil": "6.1.1",
    "pygments": "2.18.0",
    "pymdown-extensions": "10.13",
    "pyyaml": "6.0.2",
    "ruff": "0.6.9",
    "starlette": "0.42.0",
    "tomlkit": "0.13.2",
    "typing-extensions": "4.12.2",
    "uvicorn": "0.34.0",
    "websockets": "14.1"
  },
  "Optional Dependencies": {
    "altair": "5.5.0",
    "duckdb": "1.1.3",
    "pandas": "2.2.3",
    "polars": "1.17.1",
    "pyarrow": "18.1.0"
  }
}

### Code to reproduce

```python
import polars as pl

pl.col("col_name").dt
```

**Comment by akshayka** [comment]
Thanks for reporting!

I did some digging. I've opened an issue in the Polars repository, as they've been helpful in resolving similar issues in the past: https://github.com/pola-rs/polars/issues/20496

IPython has the same issue:

<img width="418" alt="Image" src="https://github.com/user-attachments/assets/93dd2b2c-ca56-4904-b7ea-9962d9e25e4a" />

As does jedi (which marimo uses for completion).

<img width="406" alt="Image" src="https://github.com/user-attachments/assets/a44b833a-65cb-4fa4-8365-c8561f725270" />

**Comment by akshayka** [comment]
The issue I created is a duplicate of https://github.com/pola-rs/polars/issues/16933. I consider this blocked on upstream (Polars), short of us switching to a different technology to power completions.

**Comment by MarcoGorelli** [comment]
Have you considered using PyRight to do autocompletions? Using that at least this example ends autocompleting correctly

**Comment by alexander-beedie** [comment]
Good news; can confirm this upcoming fix address both IPython _and_ Marimo... 😎
https://github.com/pola-rs/polars/pull/22221

**Comment by mscolnick** [comment]
Awesome, thanks @alexander-beedie! Closing this on our end
